### ISSUE TITLE
ImageItem lut + levels combine is currently a pessimization

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#1590]
### Short description

Intent of https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/ImageItem.py#L419-L440 is to combine levels and lut such that apply levels is a no-op in the subsequent call to ```makeARGB()```

### Code to reproduce

```python
import pyqtgraph as pg
import numpy as np
pg.setConfigOptions(imageAxisOrder='row-major')

app = pg.mkQApp()
imv = pg.ImageView()
imv.show()
data = np.random.randint(0, 256, size=(4096, 4096), dtype=np.uint8)
imv.setImage(data)
app.exec_()
```

### Expected behavior

apply levels should take very little time in ```makeARGB()``` profiling

### Real behavior

both apply levels and apply lut take significant time to execute
```
> Entering functions.makeARGB
  check inputs: 0.0219 ms
  apply levels: 374.5632 ms
  apply lut: 230.2356 ms
  allocate: 0.0088 ms
  reorder channels: 84.9686 ms
  alpha channel: 11.7705 ms
< Exiting functions.makeARGB, total time: 703.3772 ms
```

### Tested environment(s)

 * PyQtGraph version: 0.11.1.dev0
 * Qt Python binding: PySide2 5.15.2 Qt 5.15.2
 * Python version: 3.7.9
 * NumPy version: 1.19.4
 * Operating system: Windows 10 x64
 * Installation method: pip install -e .

### Additional context
As far can I understand, the intended behavior relied on code that was changed in #793.

For uint8 data, 
* a lut of size 256 is created
https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/ImageItem.py#L424
* levels is automatically calculated as [0, 255]
https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/functions.py#L1121-L1122
* and value of ```scale``` _before_ #793 would be set to ```lut.shape[0] - 1``` == 255

The following lines rely on levels[1] == scale in order to bypass apply levels 
https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/functions.py#L1182-L1183

So for instance, if #793 were reverted (just for the purpose of demonstration of this issue), we would get the following:
(i.e. apply levels is practically a no-op)
```
> Entering functions.makeARGB
  check inputs: 0.0324 ms
  apply levels: 0.0243 ms
  apply lut: 215.7016 ms
  allocate: 0.0088 ms
  reorder channels: 69.7846 ms
  alpha channel: 10.7770 ms
< Exiting functions.makeARGB, total time: 298.7649 ms
```

If, on the other hand, the lines performing the combination of  lut and levels were commented out altogether, https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/ImageItem.py#L422-L440
we would get an even better timing:
i.e. both lut=None and levels=None passed into ```makeARGB()```, making both apply levels and apply lut no-ops
```
> Entering functions.makeARGB
  check inputs: 0.3414 ms
  apply levels: 0.0269 ms
  apply lut: 0.0055 ms
  allocate: 0.0036 ms
  reorder channels: 84.2574 ms
  alpha channel: 12.9225 ms
< Exiting functions.makeARGB, total time: 97.5745 ms
```

**Comment by j9ac9k** [comment]
makeARGB is a brutal method..... thanks for the diagnosis.

If I'm reading this right, it looks like there may be a few issues at play

1. the check for `maxVal != scale` which is likely not what we want
2. the calculation of effective LUT, is that even needed in this case; and if not, why are we doing extra calls to `makeARGB`

I'm happy to tinker with this method more to try and speed it up, but I feel like we need a bazillion more tests for it...

**Comment by pijyoi** [comment]
At the very least, the effective LUT code, that was originally intended as an optimization, has now become a pessimization, resulting in executing both apply levels _and_ apply lut.

Using ImageItem standalone at least allows the user to bypass _both_ apply levels and apply lut. Note that you have to supply non-default argument ```autoLevels=False``` (Note that doing the same for ImageView does not work out)

The default startup settings of VideoSpeedTest.py ends up triggering the fast path. Checking the Scale Data checkbox triggers the slow path.

```python
import pyqtgraph as pg
import numpy as np
pg.setConfigOptions(imageAxisOrder='row-major')

app = pg.mkQApp()
win = pg.GraphicsView()
win.show()
vb = pg.ViewBox()
win.setCentralItem(vb)
img = pg.ImageItem()
vb.addItem(img)

data = np.random.randint(5, 250, size=(4096, 4096), dtype=np.uint8)

# slow path
# results in makeARGB(lut=<some non-monotonic list of values>, levels=None)
# img.setImage(data)

# slow path
# results in makeARGB(lut=range(256), levels=None)
# img.setImage(data, levels=[0, 255])

# fast path
# results in makeARGB(lut=None, levels=None)
img.setImage(data, autoLevels=False)

app.exec_()
```

**Comment by pijyoi** [comment]
fixed by #1693 and #1786
